python_modules/dagster/dagster/_utils/cached_method.py

```diff
@@ -18,6 +18,8 @@ class _Sentinel:
 NO_VALUE_IN_CACHE_SENTINEL: Final = _Sentinel()
 NO_ARGS_HASH_VALUE = 0
 
+CACHED_METHOD_FIELD_SUFFIX = "_cached_method_cache"
```

Comment:
let's add a couple underscores just to make sure it doesn't conflict with legitimate user-provided name

Reply:
Let's add `"__internal__"`